kanidm_unix_int/tests/cache_layer_test.rs

```diff
@@ -514,31 +514,58 @@ fn test_cache_account_pam_allowed() {
         let fut = async move {
             cachelayer.attempt_online().await;
 
-            // Should fail
-            let a1 = cachelayer
-                .pam_account_allowed("testaccount1")
-                .await
-                .expect("failed to authenticate");
-            assert!(a1 == Some(false));
+            // ! TRACING BEGIN
+            let span = span!(Level::INFO, "test_cache_account_pam_allowed");
+            let _enter = span.enter();
+            // ? Seems like these subscribers will allow us to have json loggers
+            // ? and human-readable loggers at the same time, and select between
+            // ? them easily?
+
+            let subscriber = FmtSubscriber::builder()
+                .with_max_level(Level::INFO)
+                // .event_format(format::json()) // ! uncomment for JSON
```

Comment:
I looked around in `unix_config.rs`, but I wasn't able to figure out how I can use that to configure which formatting to do here. Are the config objects passed around? Would I need to create my own field to determine the logging type or is there an existing one?

Reply:
Once I have a field and a default, how do I actually get access to this field in functions? For example, one of the tests looks like this:

```rust
#[test]
fn test_cache_account_pam_nonexist() {
    let subscriber = FmtSubscriber::builder()
        .with_max_level(Level::INFO)
        // if I wanted JSON formatting, I would say so here
        .finish();
    // other stuff here...
}
```
I guess I'm a bit confused because there is no config objects in scope. If this isn't possible, what scenarios would I have access to the config fields?

Another thing I'm concerned about is that the formatting of the subscriber varies at a type level. How can I work this in?